Sources/CodeGen/CXX/CodeFormatter.swift

```diff
@@ -0,0 +1,88 @@
+/// Class responsible for the formatiing of the CXX code, as it is generated.
+///
+/// The callers will tell this class what tokens they want to write to the output CXX code,
+/// and this will ensure that proper formatting will be added.
+///
+/// This class will have ownership of the code that is being written.
+struct CodeFormatter: TextOutputStream {
+
+  /// The amount of indentation we apply to the code.
+  private var indentSize: Int
```

Comment:
```suggestion
  private var indentation: Int
```

For consistency with other parts of the codebase